src/player.js

```diff
@@ -1528,6 +1574,83 @@ export default class Player extends FakeEventTarget {
     }
   }
 
+  _initScrollVisibility() {
+    this._viewabilityManager.observe(
+      Utils.Dom.getElementById(this._config.ui.targetId),
+      this._config.playback.visibilityThreshold / 100,
+      this._handleScrollVisibilityChange.bind(this)
+    );
+  }
+
+  _handleScrollVisibilityChange(visible: boolean) {
+    this._isVisibleInScroll = visible;
+    this._handleVisibilityChange();
+  }
+
+  _handleVisibilityChange() {
+    const prevVisibility = this._isVisible;
+    this._isVisible = this._isTabVisible && this._isVisibleInScroll;
+
+    if (prevVisibility !== this._isVisible) {
+      this.dispatchEvent(new FakeEvent(CustomEventType.VISIBILITY_CHANGE, {visible: this._isVisible}));
+    }
+
+    if (this._config.playback.autoplay === AutoPlayType.IN_VIEW && this._isVisible && !this._playbackStart) {
+      this._autoPlay();
+    }
+  }
+
+  _initTabVisibility(): void {
+    let hiddenAttr: string;
+    let visibilityChangeEventName: string;
+    if (typeof document.hidden !== 'undefined') {
+      // Opera 12.10 and Firefox 18 and later support
+      hiddenAttr = 'hidden';
+      visibilityChangeEventName = 'visibilitychange';
+    } else if (typeof document.msHidden !== 'undefined') {
+      hiddenAttr = 'msHidden';
+      visibilityChangeEventName = 'msvisibilitychange';
+    } else if (typeof document.webkitHidden !== 'undefined') {
+      hiddenAttr = 'webkitHidden';
+      visibilityChangeEventName = 'webkitvisibilitychange';
+    }
+
+    if (hiddenAttr && visibilityChangeEventName) {
+      this._eventManager.listen(document, visibilityChangeEventName, () => {
+        this._isTabVisible = !document[hiddenAttr];
+        this.dispatchEvent(new FakeEvent(CustomEventType.TAB_VISIBILITY_CHANGE, {visible: this._isTabVisible}));
+        this._handleVisibilityChange();
+      });
+      this._isTabVisible = !document[hiddenAttr];
+    }
+  }
+
+  /**
+   * Gets the player tab visibility state
+   * @returns {boolean} - whether the browser player tab is visible
+   * @public
+   */
+  get isTabVisible(): boolean {
+    return this._isTabVisible;
+  }
+
+  /**
+   * Gets the player visibility in scroll view state
+   * @returns {boolean} - whether the player is visible in the scroll view
+   * @public
+   */
+  get isVisibleInScroll(): boolean {
```

Comment:
why do you need the getter and _isVisibleInScroll if you don't have the event SCROLL_VISIBILITY_CHANGE anymore?

Reply:
why do we need those getters? user can use events for this state changes

Reply:
the getter isVisibleInScroll is really not relevant anymore. I removed it.
The other getters are needed to get the initial state

Reply:
the initial state won't be always yes?

Reply:
Not relevant anymore